packages/protocol/engine/engine.go

```diff
@@ -346,17 +355,19 @@ func (e *Engine) initBlockStorage() {
 }
 
 func (e *Engine) initNotarizationManager() {
-	e.Consensus.BlockGadget.Events.BlockAccepted.Attach(event.NewClosure(func(block *blockgadget.Block) {
+	wp := e.Consensus.BlockGadget.Events.BlockAccepted.AttachWithNewWorkerPool(event.NewClosure(func(block *blockgadget.Block) {
 		if err := e.NotarizationManager.NotarizeAcceptedBlock(block.ModelsBlock); err != nil {
 			e.Events.Error.Trigger(errors.Errorf("failed to add accepted block %s to epoch: %w", block.ID(), err))
 		}
-	}))
-	e.Tangle.Events.BlockDAG.BlockOrphaned.Attach(event.NewClosure(func(block *blockdag.Block) {
+	}), 1)
+	e.Tangle.Events.BlockDAG.BlockOrphaned.AttachWithWorkerPool(event.NewClosure(func(block *blockdag.Block) {
 		if err := e.NotarizationManager.NotarizeOrphanedBlock(block.ModelsBlock); err != nil {
 			e.Events.Error.Trigger(errors.Errorf("failed to remove orphaned block %s from epoch: %w", block.ID(), err))
 		}
-	}))
+	}), wp)
+	e.workerPools["NotarizationManager.Blocks"] = wp
 
+	// TODO: Why is it hooked?
```

Comment:
If I recall correctly, it was initially to prevent the ledger from persisting Transactions in the statediffs out of order whenever the processed transaction spend from each other. This could be problematic in the case a transaction needs to be moved to another epoch, or deleted altogether. You have to be able to make sense of the outputs you are spending and creating in each epoch, and if such accepted transaction are not dealt with in order some inconsistencies might arise in the statediffs for each epoch: although unlikely, we wanted to play it safe!